File: backend/src/main/kotlin/BacklogZero/backend/model/Tag.kt
package BacklogZero.backend.model

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "tags")
data class Tag(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long = 0,

        @Column(name = "author_id")
        val authorId: Long,

        @Column(name = "name")
        val name: String,

        @Column(name = "created_timestamp")
        val createdTimestamp: LocalDateTime = LocalDateTime.now()
)

File: next-js/models/tag.js


class Tag {
    static _baseUrl = "/tags/";
    static get(id){}

    id;
    author;
    name;
    created_timestamp;

    constructor(id, author_id, name, created_timestamp) {
        this.id = id;
        this.author = ApiGetter(User, author_id);
        this.name = name;
        this.created_timestamp = created_timestamp;
    }
}

